File: server/dto/turn_info_dto.py
from dataclasses import dataclass

from dto.base_dto import BaseDto
from dto.cell_dto import CellDto
from dto.player_info_bundle_dto import PlayerGameInfoBundleDto


@dataclass
class TurnInfoDto(BaseDto):
    currentPlayerId: str
    isPlayer1Turn: bool
    durationInS: int
    totalTurnDurationInS: int
    notifyTurnChange: bool
    playerGameInfoBundle: PlayerGameInfoBundleDto
    updatedBoardArray: list[list[CellDto]]

File: client/src/dto/TurnInfoDto.ts
import { EMPTY_STRING } from "../env";
import { CellDto } from "./PartialCellInfoDto";
import { PlayerGameInfoBundleDto, undefinedPlayerInfoBundleDto } from "./PlayerInfoBundleDto";

export interface TurnInfoDto {
    currentPlayerId: string;
    isPlayer1Turn: boolean;
    durationInS: number;
    totalTurnDurationInS: number;
    notifyTurnChange: boolean;
    playerGameInfoBundle: PlayerGameInfoBundleDto;
    updatedBoardArray: CellDto[][];
}

export const undefinedTurnInfo: TurnInfoDto = {
    currentPlayerId: EMPTY_STRING,
    isPlayer1Turn: false,
    durationInS: 0,
    totalTurnDurationInS: 0,
    notifyTurnChange: false,
    playerGameInfoBundle: undefinedPlayerInfoBundleDto,
    updatedBoardArray: []
};
